core-api/src/main/java/com/sequenceiq/cloudbreak/doc/ModelDescriptions.java

```diff
@@ -209,6 +209,7 @@ public static class StackModelDescription {
         public static final String AMBARI_VERSION = "specific version of ambari";
         public static final String HDP_VERSION = "specific version of HDP";
         public static final String CLOUDBREAK_DETAILS = "details of the Cloudbreak that provisioned the stack";
+        public static final String SALT_CB_VERSION = "details of the salt version that was used during stack provisioning";
```

Comment:
Not really. [ClusterComponentConfigProvider.getSaltStateComponentCbVersion()](https://github.com/hortonworks/cloudbreak/blob/c27df7443800b771113bc6c9ecfa26048342b37f/cluster-api/src/main/java/com/sequenceiq/cloudbreak/cluster/service/ClusterComponentConfigProvider.java#L154-L167) retrieves the CB version stored in the Salt state cluster component. Otherwise it would not make sense to compare the Salt version with a CB version like 2.70.0.

Reply:
#14331 went into 2.70. That patch includes explicitly whitelisting the autoscale machineUser for new DH clusters. We're making a check against that version to determine if userSync was needed or not

Reply:
Okay, and is this CB version stored by the salt state component updated when the cluster is upgraded via a newer CB deployment? We essentially want to compare the latest version of CB associated with the cluster (Including cluster upgrades because upgrades will include updating salt state, which will whitelist the autoscale machineUser)

Reply:
Yes, that is my understanding as well. :smile: That version property gets also updated during a Salt state update.

Reply:
Okay, thanks! Updated this description to be more accurate! 🙂